eng/packages/http-client-csharp/generator/Azure.Generator/src/Providers/ResourceDataSerializationProvider.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Generator.CSharp.ClientModel.Providers;
+using Microsoft.Generator.CSharp.Input;
+using Microsoft.Generator.CSharp.Providers;
+using System.IO;
+
+namespace Azure.Generator.Providers
+{
+    internal class ResourceDataSerializationProvider : MrwSerializationTypeDefinition
+    {
+        public ResourceDataSerializationProvider(InputModelType inputModel, ModelProvider modelProvider) : base(inputModel, modelProvider)
+        {
+        }
+
+        protected override string BuildName() => $"{base.BuildName()}Data";
```

Comment:
Could this be handled with a visitor instead?

Reply:
same reason as https://github.com/Azure/azure-sdk-for-net/pull/47730#discussion_r1909618588